android-components/components/browser/state/src/test/java/mozilla/components/browser/state/reducer/ShoppingProductStateReducerTest.kt

```diff
@@ -0,0 +1,90 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.reducer
+
+import mozilla.components.browser.state.action.ShoppingProductAction
+import mozilla.components.browser.state.selector.findTab
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.ContentState
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.rule.MainCoroutineRule
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+
+class ShoppingProductStateReducerTest {
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule()
+
+    @Test
+    fun `WHEN shopping product action is dispatched THEN isProductUrl of the relevant tab should reflect that`() {
+        val tab1 =
+            TabSessionState(
+                id = "tab1",
+                content = ContentState(
+                    url = "https://www.amazon.com/product/123",
+                    private = false,
+                ),
+                isProductUrl = false,
+            )
+        val tab2 = TabSessionState(
+            id = "tab2",
+            content = ContentState(
+                url = "https://mozilla.org",
+                private = false,
+            ),
+            isProductUrl = false,
+        )
+        val browserState = BrowserState(tabs = listOf(tab1, tab2))
+
+        val browserStore = BrowserStore(
+            initialState = browserState,
+        )
+
+        browserStore.dispatch(
+            ShoppingProductAction.UpdateProductUrlStatusAction(tabId = "tab1", isProductUrl = true),
+        ).joinBlocking()
+
+        val actual = browserStore.state.findTab("tab1")!!.isProductUrl
+        val expected = true
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `WHEN shopping product action is dispatched THEN private tab should not be affected`() {
+        val tab1 = TabSessionState(
+            id = "tab1",
+            content = ContentState(
+                url = "https://www.amazon.com/product/123",
+                private = true,
+            ),
+            isProductUrl = false,
+        )
+        val tab2 = TabSessionState(
+            id = "tab2",
+            content = ContentState(
+                url = "https://mozilla.org",
+                private = false,
+            ),
+            isProductUrl = false,
+        )
+        val browserState = BrowserState(tabs = listOf(tab1, tab2))
+
+        val browserStore = BrowserStore(initialState = browserState)
+
+        browserStore.dispatch(
+            ShoppingProductAction.UpdateProductUrlStatusAction(tabId = "tab1", isProductUrl = true),
+        ).joinBlocking()
+
+        val actual = browserStore.state.findTab("tab1")!!.isProductUrl
+        val expected = false
```

Comment:
do you need this extra val or could it just be `assertEquals(false, actual)`

Reply:
Yeah we can simplify this 👍